ui/lib/core/addon/components/edit-form.js

```diff
@@ -42,20 +42,21 @@ export default Component.extend({
       }
       return;
     }
-    this.get('flashMessages').success(this.get(messageKey));
+    this.flashMessages.success(this.get(messageKey));
     if (this.callOnSaveAfterRender) {
       next(() => {
-        this.get('onSave')({ saveType: method, model });
+        this.onSave({ saveType: method, model });
       });
       return;
     }
-    yield this.get('onSave')({ saveType: method, model });
+    yield this.onSave({ saveType: method, model });
   })
     .drop()
     .withTestWaiter(),
 
   willDestroy() {
-    let model = this.get('model');
+    let { model } = this;
```

Comment:
my ember data knowledge is somewhat elementary, so just to make sure i'm following along here: does the model not exist after creating a namespace because `willDestroy` is called before the model is actually saved to the store?

Reply:
This one's a bit spider-y. In this case `willDestroy` is called much later, after the passed model has been cleared from the controller, which is done via the `UnloadModel` route mixin. 

First, where the component is rendered, there's an onSave action passed: 

https://github.com/hashicorp/vault/blob/3a070941525d2c8aeee4371d19ad86c6dc1a5d4d/ui/app/templates/vault/cluster/access/namespaces/create.hbs#L19

Which maps to this action on the controller: https://github.com/hashicorp/vault/blob/1a0ad926a5b1ec4ae4f62a9d0c1cfec8ab458dc7/ui/app/controllers/vault/cluster/access/namespaces/create.js#L6-L12

So when this model saves, it's going to navigate away. That triggers the unload mixin: https://github.com/hashicorp/vault/blob/1a0ad926a5b1ec4ae4f62a9d0c1cfec8ab458dc7/ui/app/routes/vault/cluster/access/namespaces/create.js#L5

That mixin clears the current model so we don't get memory leaks and so unloading models from the store works better: https://github.com/hashicorp/vault/blob/1a0ad926a5b1ec4ae4f62a9d0c1cfec8ab458dc7/ui/app/mixins/unload-model-route.js#L17

The general case where we want the code in this hook to run is when you _haven't_ saved anything and there is a model - the code will run and rollback any changes made to the model, or if the model was newly created in the UI and not persisted to the server, calling `model.rollbackAttributes()` removes the model from the ember data store.